### PR TITLE
Users/dsinghal/update

### PR DESCRIPTION
**Pull request by dpksinghal**
PR for auto-update of agnets.

### CODE REVIEW

File: src/Misc/layoutbin/update.cmd.template [hunk 1]
```diff
@@ -9,6 +9,7 @@ rem  EXISTAGENTVERSION = 2.100.0
 rem  DOWNLOADAGENTVERSION = 2.101.0 
 rem  UPDATELOG = _diag/SelfUpdate-UTC.log
 rem  RESTARTINTERACTIVEAGENT = 0/1
+rem  ISINTERACTIVESESSIONCONFIGURED = 0/1
```
Comment (dpksinghal) [inline src/Misc/layoutbin/update.cmd.template]: when it runs as service, I don't see any sync between script laying out the files and the agent service starting the agent.listener.exe again. i'll confirm this with Ting.

File: src/Agent.Service/Windows/AgentListener.cs [hunk 2]
```diff
@@ -238,6 +238,8 @@ private string GetDiagnosticFolderPath()
 
         private Process CreateAgentListener()
         {
+            var pId = Process.GetCurrentProcess().Id;
```
Comment (dpksinghal) [inline src/Agent.Service/Windows/AgentListener.cs]: >GetCurrentProcess [](start = 30, length = 17)

remove this.

File: src/Agent.Listener/SelfUpdater.cs [hunk 3]
```diff
@@ -363,11 +364,23 @@ private void DeletePreviousVersionAgentBackup(CancellationToken token)
...
+            bool isInteractiveSessionConfigured = false;
 #if OS_WINDOWS
             string templateName = "update.cmd.template";
+            
+            var interactiveSessionMgr = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+            isInteractiveSessionConfigured = interactiveSessionMgr.IsInteractiveSessionConfigured();
+            if(isInteractiveSessionConfigured)
+            {
+                Trace.Info($"Interactive session is configured.");             
+                processName = $"AgentService{IOUtil.ExeExtension}";
```
Comment (dpksinghal) [inline src/Agent.Listener/SelfUpdater.cs]: >processName [](start = 16, length = 11)

dont change.

File: src/Agent.Listener/SelfUpdater.cs [hunk 4]
```diff
@@ -376,12 +389,13 @@ private string GenerateUpdateScript(bool restartInteractiveAgent)
             string template = File.ReadAllText(templatePath);
 
             template = template.Replace("_PROCESS_ID_", processId.ToString());
-            template = template.Replace("_AGENT_PROCESS_NAME_", $"Agent.Listener{IOUtil.ExeExtension}");
+            template = template.Replace("_AGENT_PROCESS_NAME_", processName);
```
Comment (dpksinghal) [inline src/Agent.Listener/SelfUpdater.cs]: >processName [](start = 64, length = 11)

use the process name.

**Comment by ericsciple** [comment]
I'm confused by all the PRs. What is this PR for? Should this one be closed?

**Comment by dpksinghal** [comment]
@ericsciple 
You can ignore this PR. This is the auto-update work that I was working upon until yesterday meeting.

So far I have been working in the following mode:

I work in a private branch and then push the updates to the feature branch. Once the changes are good and reviewed internally, I engage you and Ting while generating the PR against master. So you can ignore the PRs if they are not against master.

Please let me know if you have other better ways to deal with this.

Thank you,
Deepak.

**Pull request closed without merging**